lib/events/impl/agent.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Events.Impl.Agent do
+  @moduledoc false
+  # Dummy implementation, not really performant but good for designing the APIs.
+
+  @behaviour Events.Impl.Callbacks
+
+  @impl true
+  def child_spec(_) do
+    %{
+      id: __MODULE__,
+      start: {Agent, :start_link, [fn -> %{} end, [name: __MODULE__]]}
+    }
+  end
+
+  @impl true
+  def subscribe(sub_id, prefix, module, function, config) do
+    Agent.get_and_update(__MODULE__, fn subs ->
+      if Map.has_key?(subs, sub_id) do
+        {:error, subs}
+      else
+        {:ok, Map.put(subs, sub_id, {sub_id, prefix, module, function, config})}
+      end
+    end)
+  end
+
+  @impl true
+  def unsubscribe(sub_id) do
+    Agent.get_and_update(__MODULE__, fn subs ->
+      if Map.has_key?(subs, sub_id) do
+        {:ok, Map.delete(subs, sub_id)}
+      else
+        {:error, subs}
+      end
+    end)
+  end
+
+  @impl true
+  def list_subscribed_to(event) do
```

Comment:
This functions seems not to be (at least directly) tested.

Reply:
Yes, intentionally. We test `emit/3` which uses it to get all the subscribers. Do you feel like it should be tested directly?

Reply:
I didn't notice that it's public because it's a callback, and it's not public in the `Events` itself.
I don't think it's necessary then